src/py/flwr/client/node_state.py

```diff
@@ -15,22 +15,25 @@
 """Node state."""
 
 
-from typing import Any, Dict
+from typing import Any, Dict, Optional
 
 from flwr.common import Context, RecordSet
 
 
 class NodeState:
     """State of a node where client nodes execute runs."""
 
-    def __init__(self) -> None:
+    def __init__(self, partition_id: Optional[int] = None) -> None:
```

Comment:
Better to set the default only in one place and then force setting the value everywhere else:
```suggestion
    def __init__(self, partition_id: Optional[int]) -> None:
```